File: backend/src/main/java/com/gestion/reservas/service/NotificacionService.java
package com.gestion.reservas.service;

import com.gestion.reservas.entity.Notificacion;
import com.gestion.reservas.entity.Usuario;

import java.util.List;

public interface NotificacionService {

    List<Notificacion> findByUsuarioId(Long idUsuario);

    void crearYEnviarNotificacion(Usuario usuario, String mensaje);
    void marcarComoLeida(Long idNotificacion);
}

File: frontend/src/app/core/services/notificacion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Notificacion } from '../models/notificacion.model';

@Injectable({
  providedIn: 'root',
})
export class NotificacionService {
  private apiUrl = `${environment.apiUrl}/api/notificaciones`;

  constructor(private http: HttpClient) {}

  getNotificacionesUsuario(idUsuario: number): Observable<Notificacion[]> {
    return this.http.get<Notificacion[]>(`${this.apiUrl}/usuario/${idUsuario}`);
  }

  marcarComoLeida(idNotificacion: number): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${idNotificacion}/leida`, {});
  }
}
